src/test/shared/logger/outputChannelLogger.test.ts

```diff
@@ -1,43 +1,82 @@
 /*!
- * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import * as assert from 'assert'
 import { OutputChannelTransport, MESSAGE } from '../../../shared/logger/outputChannelTransport'
 import { MockOutputChannel } from '../../mockOutputChannel'
 
-describe('OutputChannelTransport', async () => {
+describe('OutputChannelTransport', async function () {
     let outputChannel: MockOutputChannel
+    const ansiCode = '\u001B[31m'
+    const loggedMessage1 = 'This is my logged message1'
+    const loggedMessage2 = 'This is my logged message2'
 
-    beforeEach(async () => {
+    beforeEach(async function () {
         outputChannel = new MockOutputChannel()
     })
 
-    it('logs content', async () => {
-        const loggedMessage = 'This is my logged message'
+    it('logs content', async function () {
+        await runTests(
+            [loggedMessage1, `${ansiCode}${loggedMessage2}`],
+            [loggedMessage1, `${ansiCode}${loggedMessage2}`],
+            false
+        )
+    })
+
+    it('strips ANSI codes from logged content (if told to)', async function () {
+        await runTests([loggedMessage1, `${ansiCode}${loggedMessage2}`], [loggedMessage1, loggedMessage2], true)
+    })
+
+    async function runTests(logStrings: string[], outputText: string[], stripAnsi: boolean, timeout: number = 1000) {
+        assert.ok(logStrings.length === outputText.length, 'Inputs are uneven')
+
+        const targetMessages = new Map<string, (value: void | PromiseLike<void>) => void>()
         const transport = new OutputChannelTransport({
             outputChannel,
+            stripAnsi,
         })
 
         // OutputChannel is logged to in async manner
-        const waitForLoggedText = new Promise<void>(resolve => {
-            outputChannel.onDidAppendText(loggedText => {
-                if (loggedText === loggedMessage) {
-                    resolve()
-                }
-            })
+        outputChannel.onDidAppendText(loggedText => {
+            const res = targetMessages.get(loggedText)
+            if (res) {
+                targetMessages.delete(loggedText)
+                res()
+            } else {
+                assert.ok(false, 'Logged unknown message')
+            }
         })
 
-        transport.log(
-            {
-                level: 'info',
-                message: loggedMessage,
-                [MESSAGE]: loggedMessage,
-            },
-            async () => {
-                // Test will timeout if expected text is not encountered
-                await waitForLoggedText
-            }
```

Comment:
Note: this test did nothing! This was plowed through and was never checked (a `console.log` statement in the `waitForLoggedText` function would fire _after_ the test passed successfully, regardless of what happens there)